### PR TITLE
Optimize creation of new payloads

### PR DESCRIPTION
**Pull request by turbolent**
At the end of the migration, registers are converted back to payloads.

This effectively doubles the memory usage, as the key-value pairs are stored both in the registers and in the newly created payloads.

Reduce peak memory usage, by allowing the garbage collection of account registers that have already gotten converted to payloads.

This change brings peak memory usage back to similar, and at times even lower, levels before #5897.

Also optimize speed by creating payloads in parallel.

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

File: cmd/util/ledger/util/registers/registers.go [hunk 1]
```diff
@@ -184,24 +230,26 @@ func (a *AccountRegisters) Owner() string {
...
-	payloads := make([]*ledger.Payload, 0, len(a.registers))
-	return a.appendPayloads(payloads)
+	payloads := make([]*ledger.Payload, 0, a.Count())
+	a.insertPayloads(payloads)
+	return payloads
 }
 
-func (a *AccountRegisters) appendPayloads(payloads []*ledger.Payload) []*ledger.Payload {
+func (a *AccountRegisters) insertPayloads(payloads []*ledger.Payload) {
+	index := 0
```
Comment (fxamacker) [inline cmd/util/ledger/util/registers/registers.go:239]: Maybe check if `len(payloads) == len(a.registers)` before `for` loop.

File: cmd/util/ledger/util/registers/registers.go [hunk 2]
```diff
@@ -84,11 +84,53 @@ func (b *ByAccount) ForEach(f ForEachCallback) error {
...
+	payloads := make([]*ledger.Payload, b.Count())
+
+	type job struct {
+		registers *AccountRegisters
+		payloads  []*ledger.Payload
+	}
+
+	var wg sync.WaitGroup
+
+	jobs := make(chan job)
```
Comment (fxamacker) [inline cmd/util/ledger/util/registers/registers.go:97]: Maybe use a buffered channel.

**Review by fxamacker — APPROVED** [review]
Nice!  I only left couple of minor comments.

File: cmd/util/ledger/util/registers/registers.go (hunk 1 above)
Comment (turbolent) [reply cmd/util/ledger/util/registers/registers.go:249]: Good idea! Added in 87ff7a2027fb45c84a1d7dc46fcce1084e05ba65

File: cmd/util/ledger/util/registers/registers.go (hunk 2 above)
Comment (turbolent) [reply cmd/util/ledger/util/registers/registers.go:97]: Does it make any difference?

Comment (turbolent) [reply cmd/util/ledger/util/registers/registers.go]: Switched to a buffered channel here and in the other recently added channels: [6b423c3](https://github.com/onflow/flow-go/pull/5914/commits/6b423c330f5c1fb831eba8208d8309c7ce91e81c)

**Pull request merged**